daemon/daemon.go

```diff
@@ -82,21 +79,22 @@ func (d *Daemon) Main(serve func(string, string)) int {
 // client.
 func (d *Daemon) Spawn() error {
 	binPath := d.BinPath
-	// Determine binPath.
-	if binPath == "" {
-		if len(os.Args) > 0 && path.IsAbs(os.Args[0]) {
-			binPath = os.Args[0]
-		} else {
-			// Find elvish in PATH
-			paths := strings.Split(os.Getenv("PATH"), ":")
-			result, err := util.Search(paths, "elvish")
-			if err != nil {
-				return errors.New("cannot find elvish: " + err.Error())
-			}
-			binPath = result
+	if !filepath.IsAbs(binPath) {
+		var err error
+		binPath, err = exec.LookPath(os.Args[0])
+		if err != nil {
+			return errors.New("cannot find elvish: " + err.Error())
 		}
 	}
-	return forkExec(nil, 0, binPath, d.DbPath, d.SockPath, d.LogPathPrefix)
+
+	cmd := exec.Command(binPath, "-daemon",
+		"-forked", "0",
+		"-bin", binPath,
+		"-db", d.DbPath,
+		"-sock", d.SockPath,
+		"-logprefix", d.LogPathPrefix,
+	)
+	return cmd.Run()
```

Comment:
> I think it's better to understand what is happening under the scene than using a different interface to do the job :)

Agreed. The point here is, trying not to re-invent the wheel by using high level API when possible since it is not recommended to use "syscall" package directly according to its doc.

A quick gdb showed the daemon was stuck during the syscall, which remind me to check the flags done internally by `exec.Command`.

```
(gdb) bt
#0  syscall.Syscall () at /usr/lib/go-1.8/src/syscall/asm_linux_amd64.s:27
#1  0x000000000047e516 in syscall.readlen (fd=1, p=0xc420125ae0 "", np=8, n=13, err=...) at /usr/lib/go-1.8/src/syscall/zsyscall_linux_amd64.go:1085
#2  0x0000000000478313 in syscall.forkExec (argv0=..., argv=..., attr=0xc420125da0, pid=10, err=...) at /usr/lib/go-1.8/src/syscall/exec_unix.go:202
#3  0x0000000000478754 in syscall.ForkExec (argv0=..., argv=..., attr=0xc420125da0, pid=140729429880769, err=...)
    at /usr/lib/go-1.8/src/syscall/exec_unix.go:235
#4  0x0000000000541bfa in github.com/elves/elvish/daemon.forkExec (attr=0xc420125da0, forkLevel=1, binPath=..., dbPath=..., sockPath=..., logPathPrefix=...,
    ~r6=...) at /home/go/src/github.com/elves/elvish/daemon/daemon.go:120
#5  0x0000000000541a78 in github.com/elves/elvish/daemon.(*Daemon).pseudoFork (d=0xc4201145f0, attr=0xc420125da0, ~r1=842351593000)
    at /home/go/src/github.com/elves/elvish/.build/src/github.com/elves/elvish/daemon/daemon.go:104
#6  0x000000000054173a in github.com/elves/elvish/daemon.(*Daemon).Main (d=0xc4201145f0, serve={void (struct string, struct string)} 0xc420125e48,
    ~r1=14689824) at /home/go/src/github.com/elves/elvish/daemon/daemon.go:67
#7  0x0000000000816869 in main.main () at /home/go/src/github.com/elves/elvish/main.go:121
```